workflow/controller/operator.go

```diff
@@ -2476,6 +2476,17 @@ func (woc *wfOperationCtx) computeMetrics(metricList []*wfv1.Prometheus, localSc
 	}
 }
 
+func (woc *wfOperationCtx) reportMetricEmissionError(errorString string) {
+	woc.wf.Status.Conditions.UpsertConditionMessage(
+		wfv1.WorkflowCondition{
```

Comment:
tests?

Reply:
I would say adding a test for this is unnecessary. It simply appends a string in a list element.

Reply:
is it hard to test?

Reply:
Quite the opposite. In my opinion a test for this simple behavior would be too trivial to warrant itself, but I am open to other opinions.

Regardless, I added a test for this.